--- a/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldELF.h  2026-01-10 23:06:00
+++ b/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldELF.h	2026-01-10 23:06:59
@@ -13,6 +13,8 @@
 #ifndef LLVM_LIB_EXECUTIONENGINE_RUNTIMEDYLD_RUNTIMEDYLDELF_H
 #define LLVM_LIB_EXECUTIONENGINE_RUNTIMEDYLD_RUNTIMEDYLDELF_H
 
+#include <vector>
+
 #include "RuntimeDyldImpl.h"
 #include "llvm/ADT/DenseMap.h"
 
@@ -153,7 +155,7 @@
   // When a module is loaded we save the SectionID of the EH frame section
   // in a table until we receive a request to register all unregistered
   // EH frame sections with the memory manager.
-  SmallVector<SID, 2> UnregisteredEHFrameSections;
+  std::vector<SID> UnregisteredEHFrameSections;
 
   // Map between GOT relocation value and corresponding GOT offset
   std::map<RelocationValueRef, uint64_t> GOTOffsetMap;
