--- a/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldELF.cpp	2026-01-12 16:00:24
+++ b/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldELF.cpp	2026-01-13 12:28:50
@@ -228,10 +228,12 @@
 void RuntimeDyldELF::registerEHFrames() {
   for (int i = 0, e = UnregisteredEHFrameSections.size(); i != e; ++i) {
     SID EHFrameSID = UnregisteredEHFrameSections[i];
-    uint8_t *EHFrameAddr = Sections[EHFrameSID].getAddress();
-    uint64_t EHFrameLoadAddr = Sections[EHFrameSID].getLoadAddress();
-    size_t EHFrameSize = Sections[EHFrameSID].getSize();
-    MemMgr.registerEHFrames(EHFrameAddr, EHFrameLoadAddr, EHFrameSize);
+    if (EHFrameSID < Sections.size()) {
+      uint8_t *EHFrameAddr = Sections[EHFrameSID].getAddress();
+      uint64_t EHFrameLoadAddr = Sections[EHFrameSID].getLoadAddress();
+      size_t EHFrameSize = Sections[EHFrameSID].getSize();
+      MemMgr.registerEHFrames(EHFrameAddr, EHFrameLoadAddr, EHFrameSize);
+    }
   }
   UnregisteredEHFrameSections.clear();
 }
